scrapy_poet/utils/__init__.py

```diff
@@ -1,16 +1,61 @@
+import os
+from inspect import isasyncgenfunction
 from typing import Dict
 from unittest import mock
 
-from pytest_twisted import inlineCallbacks
+from scrapy import signals
 from scrapy.crawler import Crawler
 from scrapy.exceptions import CloseSpider
+from scrapy.http import Request, Response
+from scrapy.settings import Settings
+from scrapy.utils.project import inside_project, project_data_dir
 from scrapy.utils.python import to_bytes
 from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
 from twisted.internet.task import deferLater
 from twisted.web.resource import Resource
 from twisted.web.server import NOT_DONE_YET
+from web_poet import HttpRequest, HttpResponse, HttpResponseHeaders
 
-from tests.mockserver import MockServer
+
+def get_scrapy_data_path(createdir: bool = True, default_dir: str = ".scrapy") -> str:
+    """Return a path to a folder where Scrapy is storing data.
+
+    Usually that's a .scrapy folder inside the project.
+    """
+    # This code is extracted from scrapy.utils.project.data_path function,
+    # which does too many things.
+    path = project_data_dir() if inside_project() else default_dir
+    if createdir:
+        os.makedirs(path, exist_ok=True)
+    return path
+
+
+def http_request_to_scrapy_request(request: HttpRequest, **kwargs) -> Request:
+    return Request(
+        url=str(request.url),
+        method=request.method,
+        headers=request.headers,
+        body=request.body,
+        **kwargs,
+    )
+
+
+def scrapy_response_to_http_response(response: Response) -> HttpResponse:
+    """Convenience method to convert a ``scrapy.http.Response`` into a
+    ``web_poet.HttpResponse``.
+    """
+    kwargs = {}
+    encoding = getattr(response, "_encoding", None)
+    if encoding:
+        kwargs["encoding"] = encoding
+    return HttpResponse(
+        url=response.url,
+        body=response.body,
+        status=response.status,
+        headers=HttpResponseHeaders.from_bytes_dict(response.headers),
+        **kwargs,
+    )
 
 
 class HtmlResource(Resource):
```

Comment:
I wonder if most of the content that was in tests/utils.py should be in its own submodule, e.g. `utils.testing`. For example, you would not expect the same from `utils.create_scrapy_settings` that from `utils.testing.create_scrapy_settings`.

Reply:
Makes sense. Updated in https://github.com/scrapinghub/scrapy-poet/pull/88/commits/5d2f0f9d081780340467f9f9bdc430feb7df4272. cc @wRAR